state/assign_test.go

```diff
@@ -821,90 +840,162 @@ var assignUsingConstraintsTests = []struct {
 	assignOk                bool
 }{
 	{
-		unitConstraints:         "",
-		hardwareCharacteristics: "",
+		// 0
+		unitConstraints:         "arch=amd64",
+		hardwareCharacteristics: "arch=amd64",
 		assignOk:                true,
 	}, {
+		// 1
 		unitConstraints:         "arch=amd64",
 		hardwareCharacteristics: "none",
 		assignOk:                false,
 	}, {
+		// 2
 		unitConstraints:         "arch=amd64",
 		hardwareCharacteristics: "cores=1",
 		assignOk:                false,
 	}, {
-		unitConstraints:         "arch=amd64",
-		hardwareCharacteristics: "arch=amd64",
-		assignOk:                true,
-	}, {
+		// 3
 		unitConstraints:         "arch=amd64",
 		hardwareCharacteristics: "arch=i386",
 		assignOk:                false,
 	}, {
+		// 4
 		unitConstraints:         "mem=4G",
 		hardwareCharacteristics: "none",
 		assignOk:                false,
 	}, {
+		// 5
 		unitConstraints:         "mem=4G",
 		hardwareCharacteristics: "cores=1",
 		assignOk:                false,
 	}, {
+		// 6
+		unitConstraints:         "arch=amd64 mem=4G",
+		hardwareCharacteristics: "arch=amd64 mem=4G",
+		assignOk:                true,
+	}, {
+		// 7
 		unitConstraints:         "mem=4G",
 		hardwareCharacteristics: "mem=4G",
-		assignOk:                true,
+		assignOk:                false,
+	}, {
+		// 8
+		unitConstraints:         "arch=amd64 mem=4G",
+		hardwareCharacteristics: "arch=amd64 mem=2G",
+		assignOk:                false,
 	}, {
+		// 9
 		unitConstraints:         "mem=4G",
 		hardwareCharacteristics: "mem=2G",
 		assignOk:                false,
 	}, {
+		// 10
+		unitConstraints:         "arch=amd64 cores=2",
+		hardwareCharacteristics: "arch=amd64 cores=2",
+		assignOk:                true,
+	}, {
+		// 11
 		unitConstraints:         "cores=2",
 		hardwareCharacteristics: "cores=2",
-		assignOk:                true,
+		assignOk:                false,
```

Comment:
So where we change these to false, this is due to what we discussed previously?

The machine will not be a candidate for distribution now because we can't determine whether is satisfies arch requirements?

Reply:
This requires the arch otherwise it can't satisfy the assignment. So the fact that it did previously and not now is a sign of a behavioral change, the fix and as seen in other tests is that adding `arch=amd64` and then assignments are satisified.